src/app/core/services/notifications/fcm-rest-notification.service.ts

```diff
@@ -0,0 +1,186 @@
+
+import {
+  DefaultMaxUpstreamResends,
+  DefaultPackageName,
+  DefaultSourcePrefix
+} from '../../../../assets/data/defaultConfig'
+import { SingleNotification } from '../../../shared/models/notification-handler'
+import { TaskType } from '../../../shared/utilities/task-type'
+import { getSeconds } from '../../../shared/utilities/time'
+import { RemoteConfigService } from '../config/remote-config.service'
+import { SubjectConfigService } from '../config/subject-config.service'
+import { LocalizationService } from '../misc/localization.service'
+import { LogService } from '../misc/log.service'
+import { ScheduleService } from '../schedule/schedule.service'
+import { StorageService } from '../storage/storage.service'
+import { FcmNotificationService } from './fcm-notification.service'
+import { NotificationGeneratorService } from './notification-generator.service'
+
+@Injectable()
+export class FcmRestNotificationService extends FcmNotificationService {
+  private APP_SERVER_URL = 'http://localhost:8080'
+  apiClient
+
+  constructor(
+    public notifications: NotificationGeneratorService,
+    public storage: StorageService,
+    public schedule: ScheduleService,
+    public config: SubjectConfigService,
+    public firebase: Firebase,
+    public platform: Platform,
+    public logger: LogService,
+    public remoteConfig: RemoteConfigService,
+    public localization: LocalizationService
+  ) {
+    super(storage, config, firebase, platform, logger, remoteConfig)
+    this.initApiClient()
+  }
+
+  async initApiClient() {
+    await Swagger({ url: `${this.APP_SERVER_URL}/v3/api-docs` }).then(
+      client => {
+        this.apiClient = client
+      }
+    )
+  }
+
+  getSubjectDetails() {
+    return this.checkProjectAndSubjectExistElseCreate()
+  }
+
+  publishAllNotifications(user, sourceId, limit): Promise<any> {
+    return this.schedule.getTasks(TaskType.ALL).then(tasks => {
+      const fcmNotifications = this.notifications
+        .futureNotifications(tasks, limit)
+        .map(t => this.format(t, sourceId))
+      this.logger.log('NOTIFICATIONS Scheduling FCM notifications')
+      this.logger.log(fcmNotifications)
+      return Promise.all(
+        fcmNotifications
+          .map(n => this.sendNotification(n, user.subjectId, user.projectId))
+          .concat([this.setLastNotificationUpdate()])
+      )
+    })
+  }
+
+  publishTestNotification(user, sourceId): Promise<void> {
+    return this.sendNotification(
+      this.format(this.notifications.createTestNotification(), sourceId),
+      user.subjectId,
+      user.projectId
+    )
+  }
+
+  cancelAllNotifications(user): Promise<any> {
+    return this.apiClient.apis[
+      'fcm-notification-controller'
+    ].deleteNotificationsForUser({
+      subjectId: user.subjectId,
+      projectId: user.projectId
+    })
+  }
+
+  cancelSingleNotification(user, notificationId) {
+    return this.apiClient.apis['fcm-notification-controller']
```

Comment:
it would be better to create constants for these api names at the top of the class. So easy to change if we change it in the api. Perhaps can also be taken from the remote config if present.

Reply:
Made the change.